articles/machines/resources-estimator.md

```diff
@@ -0,0 +1,149 @@
+instance with two columns: `Metric` and `Sum`, indexed by the metrics names.
+
+The following code shows how to retrieve and print the total number of `QubitClifford`, `T` and `CNOT` 
+gates used by a Q# operation:
+
+```csharp
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+namespace Quantum.MyProgram
+{
+    class Driver
+    {
+        static void Main(string[] args)
+        {
+            ResourcesEstimator estimator = new ResourcesEstimator();
+            var res = MyQuantumProgram.Run(estimator).Result;
+
+            var data = estimator.Data;
+            Console.WriteLine($"QubitCliffords: {data.Rows.Find("QubitClifford")["Sum"]}");
+            Console.WriteLine($"Ts: {data.Rows.Find("T")["Sum"]}");
+            Console.WriteLine($"CNOTs: {data.Rows.Find("CNOT")["Sum"]}");
+        }
+    }
+}
+```
+
+## Metrics Reported
+
+The following is the list of metrics estimated by the `ResourcesEstimator`:
+
+* __CNOT__: The count of CNOT (also known as the Controlled Pauli X gate) gates executed.
+* __QubitClifford__: The count of any single qubit Clifford and Pauli gate executed.
+* __Measure__:  The count of any measurement executed.
+* __R__: The count of any single qubit rotation excluding T, Clifford and Pauli gates executed.
+* __T__: The count of T gate and its conjugates, including the T gate, T_x = H.T.H, and T_y = Hy.T.Hy, executed.
+* __Depth__: Depth of the quantum circuit executed by the Q# operation.
+* __Width__: Maximum number of qubits allocated during the execution of the Q# operation.
+* __BorrowedWith__: Maximum number of qubits borrowed inside the Q# operation.
+
+
+## Providing the Probability of Measurement Outcomes
+
+<xref:microsoft.quantum.primitive.assertprob> from the <xref:microsoft.quantum.primitive> namespace can 
+be used to provide information about the expected probability of a measurement to help drive the execution 
+of the Q# program. The following example illustrates this:
+
+```qsharp
+operation Teleportation (source : Qubit, target : Qubit) : Unit {
+    body (...) {
+        using (ancillaRegister = Qubit[1]) {
+            let ancilla = ancillaRegister[0];
+
+            H(ancilla);
+            CNOT(ancilla, target);
+
+            CNOT(source, ancilla);
+            H(source);
+
+            AssertProb([PauliZ], [source], Zero, 0.5, "Outcomes must be equally likely", 1e-5);
+            AssertProb([PauliZ], [ancilla], Zero, 0.5, "Outcomes must be equally likely", 1e-5);
+
+            if (M(source) == One)  { Z(target); X(source); }
+            if (M(ancilla) == One) { X(target); X(ancilla); }
+        }
+    }
+}
+```
+
+When the `ResourcesEstimator` encounters `AssertProb` it will record that measuring
+`PauliZ` on `source` and `ancilla` should be given an outcome of `Zero` with probability
+0.5. When the simulator executes `M` later, it will find the recorded values of
```

Comment:
Which simulator is it? We have more than one, so need to clarify this

Reply:
It's the same estimator, so just replaced it with _it_.